homeassistant/components/tado/water_heater.py

```diff
@@ -0,0 +1,303 @@
+"""
+Support for Tado hot water zones.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/water_heater/tado/
+"""
+import logging
+
+from homeassistant.components.water_heater import (
+    SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    WaterHeaterDevice,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import DOMAIN, SIGNAL_TADO_UPDATE_RECEIVED
+from .const import (
+    CONST_MODE_OFF,
+    CONST_MODE_SMART_SCHEDULE,
+    CONST_OVERLAY_MANUAL,
+    CONST_OVERLAY_TADO_MODE,
+    DATA,
+    TYPE_HOT_WATER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+MODE_AUTO = "auto"
+MODE_HEAT = "heat"
+MODE_OFF = "off"
+
+WATER_HEATER_MAP_TADO = {
+    "MANUAL": MODE_HEAT,
+    "TIMER": MODE_HEAT,
+    "TADO_MODE": MODE_HEAT,
```

Comment:
Should we use the available internal tado constants here, eg `CONST_OVERLAY_TADO_MODE`?

Reply:
Sure, I also changed this in `climate.py`.